app/controller/controller.js

```diff
@@ -0,0 +1,79 @@
+const database = require('../model/database.js');
+
+const getDataForTheLast12Month = async (req, res) => {
+    try {
+        const query = `SELECT * FROM warranty
+                       WHERE date_ >= CURDATE() - INTERVAL 12 MONTH`;
+
+        await database.getDataByQuery(query, function(result) {
+            res.send(result);
+        });
+    } catch (e) {
+        console.log(e);
+        throw new Error(e.message);
+    }
+}
+
+const getDataForTheCurrentMonth = async (req, res) => {
+    try {
+        const query = `SELECT * FROM warranty
+                       WHERE date_ BETWEEN DATE_FORMAT(NOW() ,'%Y-%m-01') AND NOW()`;
+
+        await database.getDataByQuery(query, function(result) {
+            res.send(result);
+        });
+    } catch (e) {
+        console.log(e);
+        throw new Error(e.message);
+    }
+}
+
+const getDataAccordingToDateSelection = async (req, res) => {
+    try {
+        const query = `SELECT * FROM warranty
+                       WHERE date_ BETWEEN ${req.body.dateFrom} AND ${req.body.dateTo}`;
+
```

Comment:
Ich vermute, dass template strings sind nicht sicher gegen SQL Injection sind. Die Daten aus dem request body könnten durch Platzhalter (?) ersetzt werden und dadurch ausserhalb des queries mitgegeben werden. 
Hier habe ich einige Beispiele gefunden:
https://evertpot.com/executing-a-mysql-query-in-nodejs/
https://mariadb.com/kb/en/getting-started-with-the-nodejs-connector/